src/main/java/edu/ie3/datamodel/models/input/system/type/evcslocation/EvcsLocationTypeUtils.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * © 2021. TU Dortmund University,
+ * Institute of Energy Systems, Energy Efficiency and Energy Economics,
+ * Research group Distribution grid planning and operation
+*/
+package edu.ie3.datamodel.models.input.system.type.evcslocation;
+
+import edu.ie3.datamodel.exceptions.ParsingException;
+import java.util.HashMap;
+
+public class EvcsLocationTypeUtils {
+
+  private static final HashMap<String, EvcsLocationType> nameToType = initMap();
+
+  private static HashMap<String, EvcsLocationType> initMap() {
+    final HashMap<String, EvcsLocationType> map = new HashMap<>(EvcsLocationType.values().length);
+    for (EvcsLocationType type : EvcsLocationType.values()) map.put(type.name(), type);
+
+    return map;
+  }
+
+  private EvcsLocationTypeUtils() {
+    throw new IllegalStateException("This is a factory class. Don't try to instantiate it.");
+  }
+
+  public static EvcsLocationType parse(String parsableString) throws ParsingException {
+    if (nameToType.containsKey(parsableString)) return nameToType.get(parsableString);
```

Comment:
I would wish for a bit less restrictive way of parsing, e.g. comparing with neglecting the case (cf. `BdewLoadProfile`). And we should consider, which case (camel, snake, ...) we actually expect. Any suggestion?

Reply:
I wrote the type names in upper & snake case because that's the Java convention regarding enums. Of course other styles are possible, we have to expect SonarLint warnings then though.